File: pattern9/take_out_v5/src/menu.py
from .node import Node
from .menu_item import MenuItem


class Menu(Node):
    def __init__(self, name: str) -> None:
        super().__init__()
        self.name = name

    def __str__(self) -> str:
        return "the {} menu is:".format(self.name)

    def addMenuItem(self, menuItem: MenuItem):
        self.addChild(menuItem)

    def addSubMenu(self, menu: "Menu"):
        self.addChild(menu)

File: pattern9/take_out_v4/src/Menu.java
package pattern9.take_out_v4.src;

public class Menu extends Node {
    private String name;

    public Menu(String name) {
        this.name = name;
    }

    public void addMenuItem(MenuItem menuItem) {
        this.addChild(menuItem);
    }

    @Override
    public String toString() {
        return "This is a " + this.name + " menu:";
    }
}
